### PR TITLE
Add camel-openapi-java wrapper

### PR DESCRIPTION
**Pull request by ozgurkisir-qlik**
Add camel-openapi-java wrapper

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -1884,29 +1884,26 @@
...
-<!--    </feature>-->
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle dependency='true'>mvn:javax.validation/validation-api/${validation-api-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-core/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson2-version}</bundle>
```
Comment (f2par0) [inline features/src/main/feature/camel-features.xml]: use jackson features instead

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -1884,29 +1884,26 @@
...
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-core/2.1.20</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-v3/2.1.20</bundle>
```
Comment (f2par0) [inline features/src/main/feature/camel-features.xml]: use $[swagger-openapi3-version}

File: features/src/main/feature/camel-features.xml [hunk 3]
```diff
@@ -1884,29 +1884,26 @@
...
+        <bundle dependency='true'>wrap:mvn:io.swagger/swagger-parser/1.0.69</bundle>
```
Comment (f2par0) [inline features/src/main/feature/camel-features.xml]: useless dependency

**Comment by f2par0** [comment]
Suggested fixes : 
```

   <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
        <feature version='${project.version}'>camel-core</feature>
        <feature>jackson</feature>
        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
        <bundle dependency='true'>mvn:javax.validation/validation-api/${validation-api-version}</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
        <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml-version}</bundle>
        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-core/${swagger-openapi3-java-parser-version}</bundle>
        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-v3/${swagger-openapi3-java-parser-version}</bundle>
        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-annotations/${swagger-openapi3-version}</bundle>
        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-models/${swagger-openapi3-version}</bundle>
        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-core/${swagger-openapi3-version}</bundle>
        <bundle dependency='true'>mvn:org.apache.commons/commons-lang3/${commons-lang3-version}</bundle>
        <bundle>mvn:org.apache.camel.karaf/camel-openapi-java/${project.version}</bundle>
    </feature>
```

**Review by f2par0 — CHANGES_REQUESTED** [review]

File: features/src/main/feature/camel-features.xml (hunk 1 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: Done

File: features/src/main/feature/camel-features.xml [hunk 4]
```diff
@@ -1942,29 +1942,23 @@
...
-<!--        <bundle dependency='true'>mvn:javax.validation/validation-api/${validation-1-api-version}</bundle>-->
-<!--        <bundle dependency='true'>wrap:mvn:io.apicurio/apicurio-data-models/${apicurio-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-annotations/${swagger-openapi3-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-models/${swagger-openapi3-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-core/${swagger-openapi3-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:org.apache.commons/commons-lang3/${commons-lang3-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-openapi-java/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
```
Comment (f2par0) [inline features/src/main/feature/camel-features.xml]: Use`  <feature version='${camel.osgi.version.range}'>camel-core</feature>`  now , (merge jbo/REFACTORE to get this property)

File: features/src/main/feature/camel-features.xml [hunk 5]
```diff
@@ -1942,29 +1942,23 @@
...
+        <feature>jackson</feature>
```
Comment (f2par0) [inline features/src/main/feature/camel-features.xml]: Use `<feature version='[2.16,2.17)'>jackson</feature>`

**Review by f2par0 — CHANGES_REQUESTED** [review]

File: features/src/main/feature/camel-features.xml (hunk 5 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

File: features/src/main/feature/camel-features.xml (hunk 4 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

**Review by f2par0 — APPROVED** [review]
LGTM

File: features/src/main/feature/camel-features.xml [hunk 6]
```diff
@@ -1498,6 +1498,22 @@
...
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-core/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-v3/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-annotations/${swagger-openapi3-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-models/${swagger-openapi3-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-core/${swagger-openapi3-version}</bundle>
+        <bundle dependency='true'>mvn:org.apache.commons/commons-lang3/${commons-lang3-version}</bundle>
+        <bundle>mvn:org.apache.camel.karaf/camel-openapi-java/${project.version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml:1516]:
```suggestion
        <bundle>mvn:org.apache.camel.karaf/camel-openapi-java/${project.version}</bundle>
    </feature>        
```
end tag is missing

File: features/src/main/feature/camel-features.xml [hunk 7]
```diff
@@ -1498,6 +1498,23 @@
         <bundle dependency='true'>mvn:io.netty/netty-codec-http/4.1.107.Final</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-olingo4/${project.version}</bundle>
     </feature>
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-core</feature>
+        <feature version='[2.16,2.17)'>jackson</feature>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
```
`camel-tooling-util` is not an OSGI bundle so `overwrite=merge` is usless

File: features/src/main/feature/camel-features.xml [hunk 8]
```diff
@@ -1498,6 +1498,23 @@
         <bundle dependency='true'>mvn:io.netty/netty-codec-http/4.1.107.Final</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-olingo4/${project.version}</bundle>
     </feature>
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-core</feature>
+        <feature version='[2.16,2.17)'>jackson</feature>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$overwrite=merge&amp;Export-Package=org.apache.camel*;version=${camel.version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
```
ditto for `camel-xml-io`

File: features/src/main/feature/camel-features.xml [hunk 9]
```diff
@@ -1498,6 +1498,23 @@
...
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle dependency='true'>mvn:javax.validation/validation-api/${validation-api-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-core/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-v3/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-annotations/${swagger-openapi3-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-models/${swagger-openapi3-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-core/${swagger-openapi3-version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Not the right artifact id, according to the camel project, it should rather be `swagger-core-jakarta`

File: features/src/main/feature/camel-features.xml [hunk 10]
```diff
@@ -1498,6 +1498,23 @@
         <bundle dependency='true'>mvn:io.netty/netty-codec-http/4.1.107.Final</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-olingo4/${project.version}</bundle>
     </feature>
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-core</feature>
+        <feature version='[2.16,2.17)'>jackson</feature>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-tooling-util/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle dependency='true'>wrap:mvn:org.apache.camel/camel-xml-io/${camel.version}$Export-Package=org.apache.camel*;version=${camel.version}</bundle>
+        <bundle dependency='true'>mvn:javax.validation/validation-api/${validation-api-version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Not the correct `validation-api`, the expected group is `jakarta.validation` and the expected version is `3.0.2`, see the dependency:tree. Use the existing feature instead

File: features/src/main/feature/camel-features.xml [hunk 11]
```diff
@@ -1498,6 +1498,23 @@
...
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
+        <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-core/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.swagger.parser.v3/swagger-parser-v3/${swagger-openapi3-java-parser-version}</bundle>
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-annotations/${swagger-openapi3-version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: Not the right artifact id, it should rather be `swagger-annotations-jakarta`

File: features/src/main/feature/camel-features.xml [hunk 12]
```diff
@@ -1498,6 +1498,23 @@
...
+        <bundle dependency='true'>mvn:io.swagger.core.v3/swagger-models/${swagger-openapi3-version}</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: ditto, not the right artifact id

File: features/src/main/feature/camel-features.xml (hunk 11 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

File: features/src/main/feature/camel-features.xml (hunk 12 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

File: features/src/main/feature/camel-features.xml (hunk 10 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

File: features/src/main/feature/camel-features.xml (hunk 9 above)
Comment (ozgurkisir-qlik) [reply features/src/main/feature/camel-features.xml]: done

File: features/src/main/feature/camel-features.xml [hunk 13]
```diff
@@ -1498,6 +1498,23 @@
         <bundle dependency='true'>mvn:io.netty/netty-codec-http/4.1.107.Final</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-olingo4/${project.version}</bundle>
     </feature>
+    <feature name='camel-openapi-java' version='${project.version}' start-level='50'>
+        <feature version='${camel.osgi.version.range}'>camel-core</feature>
+        <feature version='[2.16,2.17)'>jackson</feature>
+        <feature>jakarta-validation</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[3,4)'>jakarta-validation</feature>
```
Let's add the appropriate version range

**Review by essobedo — APPROVED** [review]
LGTM 👍

**Pull request merged**